kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/KieServerControllerAdminImpl.java

```diff
@@ -15,40 +15,54 @@
 
 import org.kie.server.api.model.KieContainerResource;
 import org.kie.server.api.model.KieContainerStatus;
+import org.kie.server.api.model.KieScannerResource;
 import org.kie.server.api.model.KieServerInfo;
-import org.kie.server.api.model.ServiceResponse;
 import org.kie.server.controller.api.KieServerControllerAdmin;
 import org.kie.server.controller.api.KieServerControllerException;
+import org.kie.server.controller.api.ModelFactory;
 import org.kie.server.controller.api.model.KieServerInstance;
 import org.kie.server.controller.api.model.KieServerInstanceInfo;
 import org.kie.server.controller.api.model.KieServerSetup;
 import org.kie.server.controller.api.model.KieServerStatus;
-import org.kie.server.controller.api.storage.KieServerControllerStorage;
-import org.kie.server.controller.api.storage.KieServerStorageAware;
-import org.kie.server.controller.impl.storage.InMemoryKieServerControllerStorage;
+import org.kie.server.controller.api.model.runtime.ServerInstanceKey;
+import org.kie.server.controller.api.model.spec.Capability;
+import org.kie.server.controller.api.model.spec.ContainerConfig;
+import org.kie.server.controller.api.model.spec.ContainerSpec;
+import org.kie.server.controller.api.model.spec.RuleConfig;
+import org.kie.server.controller.api.model.spec.ServerTemplate;
+import org.kie.server.controller.api.storage.KieServerTemplateStorage;
+import org.kie.server.controller.impl.storage.InMemoryKieServerTemplateStorage;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public abstract class KieServerControllerAdminImpl implements KieServerControllerAdmin, KieServerStorageAware {
+/**
+ * This admin api is deprecated from 6.4.x and should not be used, instead
+ * SpecManagementService should be used
+ * @see org.kie.server.controller.api.service.SpecManagementService
+ */
+@Deprecated
+public abstract class KieServerControllerAdminImpl implements KieServerControllerAdmin {
 
     private static final Logger logger = LoggerFactory.getLogger(KieServerControllerAdminImpl.class);
 
-    private KieServerControllerStorage storage = InMemoryKieServerControllerStorage.getInstance();
+
+    private KieServerTemplateStorage templateStorage = InMemoryKieServerTemplateStorage.getInstance();
 
     @Override
     public KieServerInstance addKieServerInstance(KieServerInfo kieServerInfo) throws KieServerControllerException {
-        KieServerInstance kieServerInstance = this.storage.load(kieServerInfo.getServerId());
-        if (kieServerInstance != null) {
+        ServerTemplate serverTemplate = this.templateStorage.load(kieServerInfo.getServerId());
+        if (serverTemplate != null) {
             throw new KieServerControllerException("KieServerInstance is already registered with id: " + kieServerInfo.getServerId());
         }
-        kieServerInstance = new KieServerInstance();
+        // kie server instance is just for backward compatibility
+        KieServerInstance kieServerInstance = new KieServerInstance();
```

Comment:
in general to avoid too many constructors as there are several ways to fill in the object